### ISSUE TITLE
Add support for generating getters for an enum field

### ISSUE THREAD

**Issue by marcelgarus** [marcelgarus/data_classes#8]
In a few of my projects, I have an enum that represents a class's "type" or kind (not in the Dart Type kind of way). It would be nice to let a package generate the value getters.
Take the following class:

```dart
enum Color { red, green, yellow, blue }
enum FruitShape { round, curved }

@GenerateDataClassFor()
class MutableFruit {
  @GenerateValueGetters()
  Color color;

  @GenerateValueGetters(usePrefix=true)
  FruitShape shape;

  Fruit(this.color, this.shape);
}

class Banana extends Fruit {
  Banana() : super(Color.yellow, FruitShape.curved);
}
class Blueberry extends Fruit {
  Blueberry() : super(Color.blue, FruitShape.round);
}
```

It would be cool if the `Fruit` had some getters like the following:
```dart
class Fruit {
  ...
  bool get isRed => color == Color.red;
  bool get isGreen => color == Color.green;
  ...
  bool get isShapeRound => shape == FruitShape.round;
  bool get isShapeCurved => shape == FruitShape.curved;
}
```

Why? Because then we could use some cool stuff like `Banana().isYellow`.

Feedback about this proposal is very welcome.

**Comment by marcelgarus** [comment]
You could even add an option to generate negation getters:
```dart
@GenerateValueGetters(generateNegations=true)
FruitShape shape;
```

So, you could then use `Banana().isNotRound`

**Comment by marcelgarus** [comment]
This got implemented in `3.0.0`
